src/hrflow_connectors/connectors/boards/smartrecruiters/actions.py

```diff
@@ -0,0 +1,178 @@
+from typing import Iterator, Dict, Any, Optional
+from pydantic import Field
+from ....core.action import BoardAction
+from ....core.http import HTTPStream
+from ....core.auth import SmartToken
+
+
+class SmartJobs(HTTPStream, BoardAction):
+    auth: SmartToken
+    updated_after: Optional[str] = Field(
+        None, description="custom pulling of jobs only updated after a certain date"
+    )
+    offset: int = Field(
+        ...,
+        description="if offset is 0 and limit is 10 and `totalFound` resources are 130, our response will contain the first 10",
+    )
+    posting_status: str = Field("PUBLIC", description="Job offers availability")
+    limit: int = Field(..., description="see `offset` description")
+
+    @property
+    def base_url(self):
+        return "https://api.smartrecruiters.com/jobs"
+
+    @property
+    def http_method(self):
+        return "GET"
+
+    def pull(self) -> Iterator[Dict[str, Any]]:
+        """
+        `pull` [send tokenized requests to SmartRecruiters to pull job offers content]
+
+        Returns:
+            [jobs_content]: Iterator[Dict[str, Any] [a list of jobs content in json format]
+        """
+        jobs_content = []
+        self.params["postingStatus"] = self.posting_status
+        self.params["limit"] = self.limit
+        self.params["offset"] = self.offset
+
+        if self.updated_after:
+            self.params["updatedAfter"] = self.updated_after
+
+        response = self.send_request()
+        total_found = response.json()["totalFound"] #total offers found
+        while self.offset < total_found:
+            self.params.update({"offset": self.offset})
+            response_update = self.send_request()
+            content = response_update.__dict__["content"]  #get list of job offers contents
+            jobs_content += content
+            self.offset += self.limit
```

Comment:
**It is important to avoid storing all jobs in RAM.**

The idea is to use iterators and generators to delay the calculation of the data as much as possible.

## For now, what's going on with your code is:
1. Retrieve all the jobs of each page and concatenate them in a `content` variable
2. We push one by one each job of the list

It is translated in this way:
1. RAM : `page 0 with 3 jobs`, `page 1 with 3 jobs`, ...
2. Format
3. Push jobs from page 0, 1 ...

## What we want:
1. For each page, we tell our code which function to execute.
  * Send request
  * Chain with the rest of the list (which is not stored in RAM by the way)
  * Execute `format`.
2. At push time, we call each job, we store only the page in ram and the job to transform.

This is how it works:
1. RAM: `page 0 with 3 jobs`.
2. Format
3. Push jobs from page 0
4. RAM : `page 1 with 3 jobs`
5. Format
6. Push jobs from page 1
...

I'll have to help you on this part, I'll fix it on your branch before merging.